src/controller/cna.controller/cna.controller.js

```diff
@@ -35,231 +35,249 @@ async function getCna (req, res) {
-    }
-
-    cna.name = req.body.name
-  }
-
+  const newCna = new Cna(cna)
   Cna.findOneAndUpdate()
-    .byShortName(shortName)
-    .updateOne(cna)
+    .byShortName(newCna.short_name)
+    .updateOne(newCna)
     .exec((err, result) => {
       if (err) {
         logger.error(err.stack)
         return res.status(500).send('Internal Server Error')
       }
 
       if (result.n === 0) {
-        logger.info(shortName + ' CNA was not updated in MongoDB because it does not exists in the database.')
-        return res.status(200).json({ message: 'The cna does not exist' })
-      } else {
-        logger.info(shortName + ' CNA was successfully updated in MongoDB')
+        logger.info(newCna.short_name + ' CNA could not be updated in MongoDB because it does not exist.')
+        return res.status(400).json({ message: newCna.short_name + ' CNA could not be updated in MongoDB because it does not exist.' })
       }
 
       const responseMessage = {
-        message: 'The cna was successfully updated',
-        updated: cna
+        message: 'The cna was successfully updated in MongoDB.',
+        updated: newCna
       }
 
+      logger.info('The cna was successfully updated in MongoDB.')
       return res.status(200).json(responseMessage)
     })
 }
 
 // TODO: Request body sanitation and parameter validation
-// Updates a user if one exist for the specified username. If one doesn't exists, it creates a user.
-async function updateUser (req, res) {
-  const username = req.params.id
-  const user = {}
-
-  if (req.body.username === undefined) {
-    return res.status(400).json({ message: 'Provide the user\'s username to create or update the user' })
-  } else {
-    if (typeof req.body.username !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
+// Creates a user only if the cna exist and the user doesn't exist for the specified shortname and username respectively.
+// If the user exists, it does not update the user.
+async function createUser (req, res) {
+  const cnaShortName = req.params.shortname
+  const user = req.body
 
+  if (req.body.username !== undefined) {
     user.username = req.body.username
+  } else {
+    return res.status(400).json({ message: 'The username field in the JSON body is required to create a user.' })
   }
 
-  if (req.body.cna_name !== undefined) {
-    if (typeof req.body.cna_name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    user.cna_name = req.body.cna_name
-  }
-
-  if (req.body.cna_short_name !== undefined) {
-    if (typeof req.body.cna_short_name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    user.cna_short_name = req.body.cna_short_name
+  if (req.body.cna_short_name !== undefined && req.body.cna_short_name !== cnaShortName) {
+    return res.status(400).json({ message: cnaShortName + ' URL parameter must match ' + req.body.cna_short_name + ' in the JSON body.' })
```

Comment:
Change to:

if (user.cna_short_name !== undefined && user.cna_short_name !== cnaShortName) {
    return res.status(400).json({ message: cnaShortName + ' URL parameter does not match ' + user.cna_short_name + ' in the JSON body.' })